pkg/apiserver/webhook/configmap_webhook.go

```diff
@@ -0,0 +1,118 @@
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/go-logr/logr"
+	"gopkg.in/yaml.v2"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	controllerclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	controllers "antrea.io/nephe/pkg/controllers/cloud"
+	"antrea.io/nephe/pkg/controllers/config"
+)
+
+const (
+	ConfigMap = "nephe-config"
+)
+
+// ConfigMapValidator is used to validate Config Map.
+type ConfigMapValidator struct {
+	Client  controllerclient.Client
+	Log     logr.Logger
+	decoder *admission.Decoder
+	// Contains the same address as options.ControllerConfig
+	Config       *config.ControllerConfig
+	NpController controllers.NetworkPolicyReconciler
+}
+
+// Handle handles admission request for ConfigMap
+func (v *ConfigMapValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	v.Log.V(1).Info("Received admission webhook request", "Name", req.Name, "Operation", req.Operation)
+	switch req.Operation {
+	case admissionv1.Create:
+		return v.validateCreate(req)
+	case admissionv1.Update:
+		return v.validateUpdate(req)
+	case admissionv1.Delete:
+		return v.validateDelete(req)
+	default:
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf("invalid admission webhook request"))
+	}
+}
+
+// InjectDecoder injects the decoder.
+func (v *ConfigMapValidator) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
+
+// validateCreate validates the ConfigMap create
+func (v *ConfigMapValidator) validateCreate(req admission.Request) admission.Response {
+	return admission.Allowed("")
+}
+
+// validateUpdate validates the ConfigMap and CloudResourcePrefix update
+func (v *ConfigMapValidator) validateUpdate(req admission.Request) admission.Response {
+	newConfigMap := &corev1.ConfigMap{}
+	oldConfigMap := &corev1.ConfigMap{}
+	newControllerConfig := &config.ControllerConfig{}
+	oldControllerConfig := &config.ControllerConfig{}
+	err := v.decoder.Decode(req, newConfigMap)
+	if err != nil {
+		v.Log.Error(err, "failed to decode configMap", "ConfigMapValidator", req.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	} else {
```

Comment:
Move all the code within else block to outside. We don't need an else block, as you have a return function in the if block.

Reply:
Solved